gsy_framework/data_classes.py

```diff
@@ -373,26 +429,39 @@ def csv_fields(cls) -> Tuple:
     def csv_values(self) -> Tuple:
         """Return values of class members that are needed for creation of CSV export."""
         rate = round(self.trade_rate, 4)
-        return (self.creation_time, rate, self.traded_energy, self.seller, self.buyer,
+        return (self.creation_time, rate, self.traded_energy, self.seller.name, self.buyer.name,
                 self.matching_requirements)
 
     def to_json_string(self) -> str:
         """Return json string of the representation."""
         # __dict__ instead of asdict to not recursively deserialize objects
         trade_dict = deepcopy(self.__dict__)
-        trade_dict["offer_bid"] = trade_dict["offer_bid"].to_json_string()
+        if trade_dict["match_details"]["offer"] is not None:
+            trade_dict["match_details"]["offer"] = (
+                trade_dict["match_details"]["offer"].to_json_string())
+        if trade_dict["match_details"]["bid"] is not None:
+            trade_dict["match_details"]["bid"] = (
+                trade_dict["match_details"]["bid"].to_json_string())
         if key_in_dict_and_not_none(trade_dict, "residual"):
             trade_dict["residual"] = trade_dict["residual"].to_json_string()
         if key_in_dict_and_not_none(trade_dict, "offer_bid_trade_info"):
             trade_dict["offer_bid_trade_info"] = (
                 trade_dict["offer_bid_trade_info"].to_json_string())
+        if key_in_dict_and_not_none(trade_dict, "seller"):
+            trade_dict["seller"] = trade_dict["seller"].serializable_dict()
+            if key_in_dict_and_not_none(trade_dict, "buyer"):
+                trade_dict["buyer"] = trade_dict["buyer"].serializable_dict()
```

Comment:
Why is this indented? There should always be both `seller` and `buyer` in the trade JSON, right?